src/Generation/GapicClientV2Generator.php

```diff
@@ -160,6 +161,19 @@ private function serviceName(): PhpClassMember
             ->withValue($this->serviceDetails->serviceName);
     }
 
+    private function apiVersion()
+    {
+        if (is_null($this->serviceDetails->apiVersion)) {
+            return;
+        }
+
+        return AST::property('apiVersion')
+            ->withPhpDocText('The api version of the service')
+            ->withAccess(Access::PRIVATE)
+            ->withType('string')
```

Comment:
I gave this a thought and I don't love how I am just sending a string.

I was wondering that using something similar like the line above with `Access::PRIVATE` would look better, but I wonder how we can handle different classes, specially ones that we create. Of course this is with the thought that we should support any class in order to keep it generic.

Perhaps we could create an Enum with just the basic classes for our needs?

Reply:
Like `AST::TYPE_STRING`, `AST::TYPE_ARRAY`, etc. ? Sure that makes sense to me. I agree "magic values" aren't as fun as "magic" might make them seem.

I wouldn't worry about adding constants for each known possible type just yet. Better to just add them as we need them.

Reply:
Yeah something like that.

I am gonna add that then. I like the idea of using the types we need as we need them 👍

Reply:
I ended up creating a class inside the `hasType` file to keep everything tidy as in the `Access` file. The class that contains all the constants is called `TypeDeclaration`.